src/main/java/ActionHandler.java

```diff
@@ -0,0 +1,66 @@
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+
+public class ActionHandler {
+    enum allowedTask { todo , event , deadline};
+    /**
+     * Returns an action determined by the String command. By default, if the command is not any of the keywords
+     * returns a newTaskAction
+     * @param input specifies the type of action to return
+     * @return Action
+     */
+    public Action decideAction(String input) {
+        Scanner sc = new Scanner(input);
+        String command;
+        try{
+            command = sc.next();
+        } catch (NoSuchElementException e){
+            return new EmptyAction();
+        }
+
+        Action myAction = new EmptyAction();
+        switch(command){
+            case "bye":
+                myAction = new byeAction();
+                break;
+            case "list":
+                myAction = new listAction();
+                break;
+            case "done":
+                int index = -1;
+                try{
```

Comment:
As stated earlier, there should be a space here. This is also the case for all subsequent try blocks, e.g. at line 41, 50. Remember to fix them.